open-company/open-company-auth
# Add initial Email authentication support

This PR sets the base for adding email authentication. It handles user storage and creation (via the REPL) as well as authentication and JWToken refreshing.

Because this is setting the base for a shared code base and DRYer code between API and Auth, the PR includes associated PRs in many repos:

Move shared code to OC lib:  https://github.com/open-company/open-company-lib/pull/1

Moved shared code out of API: https://github.com/open-company/open-company-api/pull/61

Add migrations to auth: https://github.com/belucid/open-company-infrastructure/pull/23

To test:
- Ensure tests are passing here, in oc.lib and in API
- Run the updated API for your tests (no functional changes, just checking for regressions)
- Create an email user at the REPL for testing.
- Use updated web app to authenticate with the user, both successfully, and not.
- Force a token refresh attempt and make sure your new token refreshes and has new expiration.
- Ensure Slack auth and refresh still works after appropriate web changes.
- Do a small set of usual operations in the web UI looking for API regressions, all good?
- Ensure deploy to staging works for auth, API and web and that you can repeat these tests steps on staging with no issues.
- After staging deploy, create a user for Stuart at the REPL on staging so he can test email auth

NB: Limit deploy to staging! Idea here is to keep PR's smaller and not massive. These PRs can be merged to mainline (staging) but should not go all the way to master (beta) until later in the week when the rest of the email cards are complete and more extensive QA is done.
